client/cached_provider.go

```diff
@@ -47,10 +47,12 @@ func CachedAPIProvider(
 	if err := cfg.validate(); err != nil {
 		return nil, err
 	}
+
 	backend, err := memory.NewBackendStore(
 		ctx,
 		cfg.apiKey, cfg.url,
 		cfg.repoKey.OwnerName, cfg.repoKey.RepoName,
+		cfg.allowHTTP,
```

Comment:
maybe we can apply the same logic here to the `ConnectSidecarProvider`, cause technically this flag could exist there too

Reply:
done